packages/client/hmi-client/src/components/widgets/tera-pdf-viewer.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+	<div class="pdf-viewer-container">
+		<div class="pages-container">
+			<template v-for="page in pages" :key="page">
+				<VuePDF
+					ref="vuePdfs"
+					class="page"
+					:pdf="pdf"
+					:page="page"
+					:fit-parent="fitToWidth"
+					:scale="scale"
+					text-layer
+					@loaded="(data) => onPageLoaded(data, page)"
+				/>
+			</template>
+		</div>
+		<div class="controls"></div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import '@tato30/vue-pdf/style.css';
+import { groupBy } from 'lodash';
+import { computed, ref, watch, useTemplateRef } from 'vue';
+import { VuePDF, usePDF } from '@tato30/vue-pdf';
+
+const DEFAULT_SCALE = 2;
+const HIGHLIGHT_DEFAULT_COLOR = 'rgba(255, 255, 0, 0.5)'; // Yellow with 50% opacity
+const BBOX_DEFAULT_COLOR = 'green';
+
+export interface PdfAnnotation {
+	pageNo: number;
+	bbox: { l: number; t: number; r: number; b: number };
+	color: string;
+	isHighlight: boolean;
+}
+
+const props = defineProps<{
+	pdfLink?: string;
```

Comment:
I think arguably pdfLink should not be nullable if you area already in the viewer?